src/Catalog/Persistence/AzureStorage.cs

```diff
@@ -309,6 +292,7 @@ protected override async Task<StorageContent> OnLoadAsync(Uri resourceUri, Cance
 
             CloudBlockBlob blob = GetBlockBlobReference(name);
 
+            await _throttle.WaitAsync();
```

Comment:
This is one of the possible download paths in Feed2Catalog. There is also an `HttpClient`-based one. Do we ever hit that path in practice? If so, do we need the throttle there too?

Reply:
I wanted to be very cautious about how I applied this throttle. I could have applied this throttle to all of the V3 jobs but I only applied it to this job because it was the only one that seemed affected by the issue that the throttling mitigates.

Similarly, I never saw Feed2Catalog fail [in the `HttpClient`-based path](https://github.com/NuGet/NuGet.Services.Metadata/blob/master/src/Catalog/PackageCatalogItemCreator.cs#L191), so I don't feel like it's necessary to add it there.

Furthermore, the `HttpClient`-based path doesn't even seem to have any retry policy. I think we would want to add that first before adding throttling--but we haven't needed either in the past, so I'm hesitant to add it with this change.

Reply:
Sounds good.